pacman/utilities/algorithm_utilities/placer_algorithm_utilities.py

```diff
@@ -68,50 +69,70 @@ def group_vertices(vertices, same_group_as_function):
         A dictionary of vertex to list of vertices that are grouped with it
     """
 
-    # Dict of vertex to list of vertices on same chip (repeated lists expected)
+    groups = create_vertices_groups(vertices, same_group_as_function)
+    # Dict of vertex to setof vertices on same chip (repeated lists expected)
+    # A empty set value indicates a set that is too big.
     same_chip_vertices = OrderedDict()
+    for group in groups:
+        for vertex in group:
+            same_chip_vertices[vertex] = group
+    for vertex in vertices:
+        if vertex not in same_chip_vertices:
+            same_chip_vertices[vertex] = {vertex}
+    return same_chip_vertices
+
+
+def add_set(all_sets, new_set):
```

Comment:
I might be being dense, but if the groups was a ordered set instead of a list, would this function become redundant, as you could just use update of the ordered set?

Reply:
Added docs for the dense.

Reply:
fine, but could it be pushed into the orderedset fort encapsulation arguments? given its basically messing with sets.